### ISSUE TITLE
Unable to render equations via KaTeX

### ISSUE THREAD

**Issue by davidzoltowski** [gwgundersen/anno#23]
I'm unable to properly render the equations in my own notes or the example note. They end up looking like this instead:

![image](https://user-images.githubusercontent.com/10839361/77965205-63e70700-7295-11ea-964f-cbaf5aaa398f.png)

I've tried installing KaTeX globally but it didn't fix the issue for me. Could you provide guidance on how to set up KaTeX to work with Anno?

**Comment by gwgundersen** [comment]
As a quick fix, you can use the `$$` and `$` delimiters for display and inline respectively. I would like to support brackets and parentheses as well, but I _think_ this is a PyPandoc issue. When the Markdown gets parsed by PyPandoc, it doesn't detect Katex unless the dollar sign delimiters are used.

If you have a lot of old notes you want to port over and this is a deal-breaker, let me know so I can prioritize. Otherwise, let's keep this open so I remember to investigate further.

**Comment by davidzoltowski** [comment]
Thanks for the quick response. I am using `$$` and `$` for equations and still getting the issue. However, I just tried installing from PyPI via `pip install anno` on another machine and the equation rendering worked out of the box. I installed from the github source on this machine, and may have made a mistake at some point. I'll see if I can fix it.

**Comment by gwgundersen** [comment]
Okay, let me know if you can't get it working from source. I uninstalled and reinstalled following the README and didn't notice an issue with the delimiters, but maybe there's another issue.

**Comment by davidzoltowski** [comment]
I think I'm encountering an issue with pypandoc on my newer machine (as everything works on the old machine). On the new machine, I reinstalled from source and still had issues with math rendering. 

In `render_markdown`,  the output of the call to `pypandoc.convert_text` was different on the two machines. For inline equations, it would include extra `\\)` and `\\)` characters
```
In [32]: render_markdown('$x$')
Out[32]: '<p><span class="math inline">\\(x\\)</span></p>\n'
```
I'm not sure why that is happening but I hacked a messy solution for now by removing `\\(`, `\\)`, `\\[`, and `\\]` characters from the output of math inline or display text in `render_markdown` with the following code: 
```
def render_markdown(value):
    extra_args = ['--katex']
    output = pypandoc.convert_text(value, to='html5', format='md',
                                   extra_args=extra_args)
    output = output.replace('"math inline">\\(','"math inline">')
    output = output.replace('\\)</span>', '</span>')
    output = output.replace('"math display">\\[','"math display">\\displaystyle')
    output = output.replace('\\]</span>', '</span>')
    return output
```
This appears to be working, although it could cause an issue down the road. 

Since I've got something that works for now, the error appears to arise from a local pypandoc issue, and I haven't been able to trace the error any further, I'll close this issue.
